ipaserver/install/httpinstance.py

```diff
@@ -305,34 +311,39 @@ def configure_certmonger_renewal_guard(self):
             if certmonger_stopped:
                 certmonger.stop()
 
-    def create_cert_db(self):
-        database = certs.NSS_DIR
-        pwd_file = os.path.join(database, 'pwdfile.txt')
-
-        for p in NSS_FILES:
-            nss_path = os.path.join(database, p)
-            ipautil.backup_file(nss_path)
-
-        # Create the password file for this db
-        password = ipautil.ipa_generate_password()
-        with io.open(pwd_file, 'w') as f:
-            f.write(password)
-
-        ipautil.run([paths.CERTUTIL, "-d", database, "-f", pwd_file, "-N"])
-
-        self.fix_cert_db_perms()
-
-    def fix_cert_db_perms(self):
-        pent = pwd.getpwnam(self.service_user)
-
-        for filename in NSS_FILES:
-            nss_path = os.path.join(certs.NSS_DIR, filename)
-            os.chmod(nss_path, 0o640)
-            os.chown(nss_path, 0, pent.pw_gid)
-            tasks.restore_context(nss_path)
+    def request_anon_keytab(self):
+        parent = os.path.dirname(paths.ANON_KEYTAB)
+        if not os.path.exists(parent):
+            os.makedirs(parent, 0o755)
+        self.run_getkeytab(self.api.env.ldap_uri, paths.ANON_KEYTAB, ANON_USER)
+
+        pent = pwd.getpwnam(IPAAPI_USER)
+        os.chmod(parent, 0o700)
+        os.chown(parent, pent.pw_uid, pent.pw_gid)
+        os.chown(paths.ANON_KEYTAB, pent.pw_uid, pent.pw_gid)
+
+    def create_password_conf(self):
+        """
+        This is the format of mod_nss pin files.
+        """
+        pwd_conf = paths.HTTPD_PASSWORD_CONF
+
+        ipautil.backup_file(pwd_conf)
+        f = open(pwd_conf, "w")
```

Comment:
This is old code moved here from `CertDB`. IMO it does not need to be refactored in this PR.